cryptographic_estimators/MREstimator/MRAlgorithms/minors.py

```diff
@@ -0,0 +1,169 @@
+from .mr_helper import minors_polynomial
+from ..mr_constants import MR_NUMBER_OF_KERNEL_VECTORS_TO_GUESS, MR_NUMBER_OF_COEFFICIENTS_TO_GUESS
+
+
+class Minors(MRAlgorithm):
+    r"""
+    Construct an instance of Minors estimator
+
+
+    INPUT:
+
+    - ``problem`` -- an instance of the MRProblem class
+    - ``w`` -- linear algebra constant (default: 3)
+    - ``theta`` -- exponent of the conversion factor (default: 2.81)
+
+    EXAMPLES::
+
+        sage: from cryptographic_estimators.MREstimator.MRAlgorithms.minors import Minors
+        sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+        sage: E = Minors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+        sage: E
+        Minors estimator for the MinRank problem with (q, m, n, k, r) = (7, 9, 10, 15, 4)
+    """
+
+    def __init__(self, problem: MRProblem, **kwargs):
+
+        super(Minors, self).__init__(problem, **kwargs)
+
+        q, m, n, k, r = self.problem.get_parameters()
+        self.set_parameter_ranges('a', 0, min(n - r, ceil(k / m)))
+        self.set_parameter_ranges('lv', 0, r)
+        self._name = "Minors"
+
+    @optimal_parameter
+    def a(self):
+        """
+        Return the optimal `a`, i.e. no. of vectors to guess in the kernel of the low-rank matrix
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.minors import Minors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: ME = Minors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: ME.a()
+            2
+
+        TESTS::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.minors import Minors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: ME = Minors(MRProblem(q=16, m=15, n=15, k=78, r=6))
+            sage: ME.a()
+            5
+        """
+        return self._get_optimal_parameter(MR_NUMBER_OF_KERNEL_VECTORS_TO_GUESS)
+
+    @optimal_parameter
+    def lv(self):
+        """
+        Return the optimal `lv`, i.e. no. of entries to guess in the solution
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.minors import Minors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: ME = Minors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: ME.lv()
+            0
+
+        TESTS::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.minors import Minors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: ME = Minors(MRProblem(q=16, m=15, n=15, k=78, r=6))
+            sage: ME.lv()
+            0
+        """
+        return self._get_optimal_parameter(MR_NUMBER_OF_COEFFICIENTS_TO_GUESS)
+
+    def _ME_time_complexity_helper_(self, m, n_reduced, k_reduced, r):
+        time = 0
```

Comment:
the only difference between `_ME_time_complexity_helper` and `_ME_memory_complexity_helper` is `w*` and `2*`. So maybe we can merge those two into one function, to reduce code complexity.